app/controllers/api/v1/products_controller.rb

```diff
@@ -47,7 +47,11 @@ def product_params
   end
 
   def check_owner
-    render_error(4111) unless product.user_id == current_user&.id
+    if current_user
+      render_error(4203) unless product.user_id == current_user&.id
+    else
+      render_error(4002)
+    end
   end
```

Comment:
where's is this used? Shouldn't it be part of a errors helper or something like that?
also probably can be called: on_error_ownership_check (since it only defines a specific error)

Reply:
It will change in the Products CRUD PR 😄